src/app/beer_garden/user.py

```diff
@@ -133,26 +133,51 @@ def user_synced_with_garden(user: User, garden: Garden) -> bool:
     except RemoteUser.DoesNotExist:
         return len(user.role_assignments) == 0
 
+    for role_assignment in user.role_assignments:
+        # If any roles that the user is assigned are not synced, the user is considered
+        # not synced
+        if not role_status.get(role_assignment.role.name, {}).get(garden.name, False):
+            return False
+
     role_assignments = UserSyncSchema().dump(user).data["role_assignments"]
 
     return role_assignments == remote_user.role_assignments
 
 
-def user_sync_status(user: User) -> dict:
-    """Provides the sync status of the provided User with each remote garden. The
-    resulting dict formatting is:
+def user_sync_status(users: List[User]) -> dict:
```

Comment:
This got reworked to take a list of users rather than an individual one.  This was done mainly because the determination of whether a user is considered synced or not now factors in whether or not all of the roles they are assigned are synced as well.  Since the the status of the roles being synced is identical for all users, I wanted to avoid having to do that check over and over for every individual user.  Instead, I changed this to take a list of users so that I can get the `role_sync_status` for all roles once and only once, no matter how many users we're checking.